easypost/services/event_service.py

```diff
@@ -56,4 +60,14 @@ def get_next_page(
         optional_params: Optional[Dict[str, Any]] = None,
     ) -> Dict[str, Any]:
         """Retrieve the next page of the list Events response."""
-        return self._get_next_page_resources(self._model_class, events, page_size, optional_params)
+        self._check_has_next_page(collection=events)
```

Comment:
question: does it make sense to put all of this into the base class so that every service can reuse it and we only override those we need (reports, shipments, etc)? That way we save on code reuse. Wouldn't that get us the best of both worlds?

Reply:
I've moved as much generic code to the base class that I feel comfortable with. The code remaining as per-service implementations is because there are/will be variance per-service (e.g. batches I know needs to be "after_id", not "before_id", the key to look for in each JSON array differs).